packages/app-builder/src/routes/_builder+/cases+/inboxes.$inboxId.tsx

```diff
@@ -69,50 +70,119 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
 }
 
 export default function Cases() {
-  const {
-    casesData: { items: cases, ...pagination },
-    filters,
-  } = useLoaderData<typeof loader>();
+  const { casesData: initialCasesData, filters: initialFilters } =
+    useLoaderData<typeof loader>();
   const inboxId = useParam('inboxId');
   const { t } = useTranslation(casesI18n);
 
+  const [casesData, setCasesData] = useState(initialCasesData);
+  const { items: cases, ...pagination } = casesData;
+
+  const [filters, setFilters] = useState(initialFilters);
+  if (initialFilters !== filters) {
+    setFilters(initialFilters);
+    setCasesData(initialCasesData);
+  }
+
+  const {
+    state: paginationState,
+    next,
+    previous,
+    reset,
+  } = useCursorPagination();
+
+  const fetcher = useFetcher<typeof loader>();
   const navigate = useNavigate();
-  const navigateCasesList = useCallback(
-    (casesFilters: CasesFilters, pagination?: PaginationParams) => {
-      navigate(
+
+  useEffect(() => {
+    if (paginationState.isPristine) {
+      return;
+    }
+
+    fetcher.submit(
+      qs.stringify(
         {
-          pathname: getRoute('/cases/inboxes/:inboxId', {
-            inboxId: fromUUID(inboxId),
-          }),
-          search: qs.stringify(
-            {
-              statuses: casesFilters.statuses ?? [],
-              dateRange: casesFilters.dateRange
-                ? casesFilters.dateRange.type === 'static'
-                  ? {
-                      type: 'static',
-                      endDate: casesFilters.dateRange.endDate || null,
-                      startDate: casesFilters.dateRange.startDate || null,
-                    }
-                  : {
-                      type: 'dynamic',
-                      fromNow: casesFilters.dateRange.fromNow,
-                    }
-                : {},
-              offsetId: pagination?.offsetId || null,
-              next: pagination?.next || null,
-              previous: pagination?.previous || null,
-            },
-            {
-              addQueryPrefix: true,
-              skipNulls: true,
-            },
-          ),
+          statuses: filters.statuses ?? [],
+          dateRange: filters.dateRange
+            ? filters.dateRange.type === 'static'
+              ? {
+                  type: 'static',
+                  endDate: filters.dateRange.endDate || null,
+                  startDate: filters.dateRange.startDate || null,
+                }
+              : {
+                  type: 'dynamic',
+                  fromNow: filters.dateRange.fromNow,
+                }
+            : {},
+          ...(paginationState.cursor
+            ? {
+                offsetId: paginationState.cursor,
+                next: true,
```

Comment:
and then if "previous" is no longer an option, perhaps we also remove the option (and the need) to pass "next" as a query param